### PR TITLE
Jw new search for dashboard

### PR DESCRIPTION
**Pull request by jwiel86**
https://www.pivotaltracker.com/story/show/138951671

### CODE REVIEW

File: app/assets/javascripts/dashboard/protocols.js.coffee [hunk 1]
```diff
@@ -167,6 +167,13 @@ $(document).ready ->
             $('.service-requests-table').on 'all.bs.table', ->
               $(this).find('.selectpicker').selectpicker()
 
+      $(document).on 'click', '.dropdown-menu li', ->
```
Comment (ghost) [inline app/assets/javascripts/dashboard/protocols.js.coffee:170]: Is it necessary that the adjacent text field is disabled when you don't select an option from the select?

File: app/models/protocol.rb [hunk 2]
```diff
@@ -49,6 +49,8 @@ class Protocol < ActiveRecord::Base
 
   has_many :principal_investigators, -> { where(project_roles: { role: %w(pi primary-pi) }) },
     source: :identity, through: :project_roles
+  has_many :authorized_users, -> { where.not(project_roles: { role: %w(pi primary-pi) }) },
```
Comment (ghost) [inline app/models/protocol.rb]: Isn't a PI technically an authorized user? I think this name could be a little confusing.

File: app/models/protocol.rb [hunk 3]
```diff
@@ -185,30 +187,46 @@ def unique_rm_id_to_protocol
...
-
-    joins(:identities).joins(:human_subjects_info).
+    exact_search_term = ActiveRecord::Base::sanitize(search_attrs[:search_text])
+
+    case search_attrs[:search_drop]
+      
+    when "PI"
+      where_clause = ["CONCAT(identities.first_name, ' ', identities.last_name) LIKE #{like_search_term} escape '!'"]
+      joins(:principal_investigators).
+        where(where_clause.compact.join(' OR ')).
```
Comment (ghost) [inline app/models/protocol.rb]: A lot of these `where_clause`'s only have one element. So invoking `.compact.join(' OR ')` has the same effect as `where_clause[0]`. Maybe you can simplify things a lot by just defining some of these `where_clause`'s as a string.

File: spec/views/dashboard/protocols/index.html.haml_spec.rb [hunk 4]
```diff
@@ -25,6 +25,7 @@
 
   before(:each) do
     assign(:user, jug2)
+    assign(:search_by, ['Authorized User', 'HR#', 'PI', 'Protocol', 'PRO#'])
```
Comment (ghost) [inline spec/views/dashboard/protocols/index.html.haml_spec.rb]: I don't think you need this anymore.

File: spec/features/dashboard/protocols/index/filters_spec.rb [hunk 5]
```diff
@@ -213,108 +224,337 @@ def visit_protocols_index_page
...
 
-      expect(@page.search_results).to have_protocols(text: "Protocol1")
-      expect(@page.search_results).to have_protocols(text: "Protocol2")
-      expect(@page.search_results).to have_no_protocols(text: "Protocol3")
+      it "should match against title case insensitively (lowercase)" do
+        visit_protocols_index_page
+        expect(@page.search_results).to have_protocols(count: 3)   
+        @page.filter_protocols.select_search(@page, "Protocol", "title")
+        @page.filter_protocols.apply_filter_button.click()
+        wait_for_javascript_to_finish
```
Comment (ghost) [inline spec/features/dashboard/protocols/index/filters_spec.rb]: How necessary is the `wait_for_javascript`? I see that the example below doesn't have it.

File: app/models/protocol.rb (hunk 2 above)
Comment (jwiel86) [reply app/models/protocol.rb]: @jayhardee9 very true.  That's how Wenjun wanted the dropdown language, so I followed that naming convention.  ['Authorized User', 'HR#', 'PI', 'Protocol', 'PRO#']

File: spec/features/dashboard/protocols/index/filters_spec.rb (hunk 5 above)
Comment (jwiel86) [reply spec/features/dashboard/protocols/index/filters_spec.rb]: you're right!  removed and passing!

**Review by Stuart-Johnson — APPROVED** [review]

**Review by jleonardw9 — APPROVED** [review]

**Comment by wtholt** [comment]
Nice work @jwiel86. My only request is if we could clean up the git history a bit?

**Review by wtholt — APPROVED** [review]

**Pull request closed without merging**